docs/contributor/migration-guide.md

```diff
@@ -8,78 +8,164 @@ It covers all necessary changes and deprecations to ensure a smooth transition.
+
+### 2.2 `module-config.yaml` Schema Comparison Example
+
+This comparison shows a generic module configuration. Replace `<module-name>`, `<channel>`, and `<version>` with your module’s actual values.
+
+##### Old Format
+
+```yaml
+name: kyma-project.io/module/<module-name>
+channel: <channel>
+version: <version>
+manifest: <module-name>-manifest.yaml
+defaultCR: <module-name>-default-cr.yaml
+annotations:
+  operator.kyma-project.io/doc-url: https://help.sap.com/.../<module-name>-module
+moduleRepo: https://github.com/kyma-project/<module-name>.git
+```
+
+##### New Format
+
+```yaml
+# modules/<module-name>/<version>/module-config.yaml
+name: kyma-project.io/module/<module-name>
+repository: https://github.com/kyma-project/<module-manager-name>.git
+version: 1.34.0
+manifest: https://github.com/kyma-project/<module-manager>/releases/download/1.34.0/<module-manager-name>.yaml
+defaultCR: https://github.com/kyma-project/<module-manager>/releases/download/1.34.0/<module-name-default-cr>.yaml
+security: sec-scanners-config.yaml
+manager:
+   name: <module-manager-name>
+   namespace: kyma-system
+   group: apps
+   version: v1
+   kind: Deployment
+associatedResources:
+   - group: operator.kyma-project.io
+     kind: <ModuleName>
+     version: v1alpha1
+   - group: operator.kyma-project.io
+     kind: LogParser
+     version: v1alpha1
+   - group: operator.kyma-project.io
+     kind: LogPipeline
+     version: v1alpha1
+   - group: operator.kyma-project.io
+     kind: MetricPipeline
+     version: v1alpha1
+   - group: operator.kyma-project.io
+     kind: TracePipeline
+     version: v1alpha1
+documentation: https://help.sap.com/docs/btp/sap-business-technology-platform/<kyma-module-name>
+icons:
+   - name: module-icon
+      # TODO: provide <module-name> icon
+     link: https://raw.githubusercontent.com/kyma-project/kyma/refs/heads/main/docs/assets/logo_icon.svg
+```
+
+### 2.3 Channel Mapping with ModuleReleaseMeta Example
+
+```yaml
+# modules/<module-name>/module-releases.yaml
+channels:
+  - channel: regular
+    version: 1.34.0
+  - channel: fast
+    version: 1.34.0
+  - channel: experimental
+    version: 1.34.0-experimental
+  - channel: dev
+    version: 1.35.0-rc1
+```
+
+On merge, the pipeline:
+
+* Validates no downgrades and version existence
+* Generates ModuleReleaseMeta CRs per landscape
+* Updates landscape-specific kustomizations to reference only active versions
+
+### 2.4 Metadata Deprecations & New Practices
+
+| Deprecated Feature                | Replacement / New Location                                                    |
+| --------------------------------- | ----------------------------------------------------------------------------- |
+| `.channel` field in module config | Moved to ModuleReleaseMeta CR (`module-releases.yaml`)                        |
+| `mandatory` on ModuleTemplate     | Set `mandatory: true` in module config; reconciler picks highest version      |
```

Comment:
Reconciler picks highest version? Is this meaning lifecycle-manager? Doesn't it try to always install the highest version?

Reply:
Also not sure if this is needed in this doc.